src/hammr/commands/image/image.py

```diff
@@ -340,149 +365,105 @@ def help_download(self):
         doParser = self.arg_download()
         doParser.print_help()
 
-
     def get_image_status(self, status):
         if (status.complete and not status.error):
-            imgStatus="Done"
+            imgStatus = "Done"
         elif status.error:
-            imgStatus="Error"
+            imgStatus = "Error"
         elif status.cancelled:
-            imgStatus="Canceled"
+            imgStatus = "Canceled"
         else:
-            imgStatus="In progress ("+str(status.percentage)+"%)"
+            imgStatus = "In progress (" + str(status.percentage) + "%)"
         return imgStatus
 
-
     def get_publish_status(self, status):
         if (status.complete and not status.error):
-            pubStatus="Done"
+            pubStatus = "Done"
         elif status.error:
-            pubStatus="Error"
+            pubStatus = "Error"
         elif status.cancelled:
-            pubStatus="Canceled"
+            pubStatus = "Canceled"
         else:
-            pubStatus="In progress ("+str(status.percentage)+"%)"
+            pubStatus = "In progress (" + str(status.percentage) + "%)"
         return pubStatus
 
     def publish_builder(self, builder, template, appliance, rInstallProfile, i, comliantImage):
+        printer.out("Publishing '" + builder["type"] + "' image (" + str(i) + "/" + str(1) + ")")
         try:
             if comliantImage is None:
-                printer.out("Publishing '"+builder["type"]+"' image ("+str(i)+"/"+str(len(template["builders"]))+")")
-
-                images = self.api.Users(self.login).Appliances(appliance.dbId).Images.Getall(Query="targetFormat.name=='"+builder["type"]+"'")
-                images = images.images.image
-                if images is None or len(images) == 0:
-                    printer.out("No images found for the template '"+template["stack"]["name"]+"' with type: "+builder["type"], printer.ERROR)
-                    return 2
+                comliantImage = self.get_image_to_publish(builder, template, appliance, i)
```

Comment:
I know that this variable already existed before your modification, but there is a typo: compliantImage